### DISCUSSION TITLE
Welcome to online battle mod Discussions!

### DISCUSSION THREAD

**Discussion by shabbywu** [shabbywu/awesome-battle-brothers#5] [Ideas]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Exorciamus** [comment]
Greetings! I am using your program bb-launcher-steam.exe to check my game's translation modification into Ukrainian. However, I am having issues with the fonts, particularly the bitmap fonts used by the game during the generation of the world map, and for the names of locations and plains. With the help of bb-launcher-steam.exe, I am able to install my own font that supports specific Ukrainian letters, and with your program, everything is displayed perfectly on the world map. However, I would like to create my own files: "cinzel_20.fnt, cinzel_20.png, cinzel_bold_20.fnt, cinzel_bold_20.png, cinzel_bold_100.fnt, cinzel_bold_100.png". But I don't understand the principles of how to do this; just using BMfont didn't work for me... So I thought maybe you could advise me on what to do, I would be very grateful for any advice or help with this!

**Reply by shabbywu** [reply]
Hi, i do have tried with BMfont. Hope it can help you.

First of all, the game logic only supports Ascii character encoding, and the game logic has **hardcoded** the alphabet size to 256.

Secondly, I remember that these `***.fnt` files describe how to map character encoding to BMFont(`***.png`). The game logic loads BMFont into array with size 256, and finds the corresponding font for rendering according to the Ascii encoding.
For example, letter **A** in Ascii encoding is **65**, the game logic use the BMFont store in index 65 to render.

If Ukrainian alphabet is compatible with Ascii encoding, it might work. If not, it's no way to do it unless patch the game logic just like ` bb-launcher` do.
![image](https://github.com/user-attachments/assets/cdace3f4-5ea1-4681-9232-8ad438e02465)

**Reply by Exorciamus** [reply]
Oh, thank you for the response, it was very informative! I doubt I can do much with it, but I heard that in ASCII you can somehow replace symbols in the encoding, for example, choose a symbol from ASCII that is not used by the game and replace it with another symbol like the letter "Ї". Although I don't really know how to implement this, I only know that the Russians somehow did this in their localization.

**Reply by shabbywu** [reply]
Yes, `replace it with another symbol` can work. for example you can map the letter "[И](https://en.wikipedia.org/wiki/%D0%98)" to letter "Ї".
> Actually, `bb-launcher` do it automatic by programming.

If you want to do it by yourself, you can follow this trip:
1. Design how letter translation from Ukrainian alphabet to ASCII.
2. Make sure all your `***.nut` are using ASCII substitute, but not Ukrainian letter.
3. Generate BMFont with ASCII, but replace to the real Ukrainian letter


Desigin your letter translation 
---
А | Б | В | Г | Ґ | Д | Е | Є | Ж | З | И | І | Ї | Й | К | Л | М | Н | О | П | Р | С | Т | У | Ф | Х | Ц | Ч | Ш | Щ | Ь | Ю | Я
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
? | ? | ? | ? | ? | ? | ? | ? | ? | ? | ? | ? | ? | ? | ? | ? | ? | ? | ? | ? | ? | ? | ? | ? | ? | ? | ? | ? | ? | ? | ? | ? | ?
